src/question/question.service.ts

```
@@ -28,11 +44,172 @@ export class QuestionService {
   constructor(
     @InjectRepository(Question) private questionRepository: Repository<Question>,
     @InjectRepository(Sentence) private sentenceRepository: Repository<Sentence>,
-    @InjectRepository(Sentence) private optionRepository: Repository<Option>,
+    @InjectRepository(Option) private optionRepository: Repository<Option>,
+    @InjectRepository(Fragment) private fragmentRepository: Repository<Fragment>,
   ) {}
 
   async getAllQuestions(): Promise<QuestionData[]> {
     const result = await this.questionRepository.find({});
     return transformToQuestionData(result);
   }
+
+  /**
+   * Validates that includeIfSelected options exist in the available options
+   * @param includeIfSelected - Array of options that should be included
+   * @param availableOptions - Array of all available options
+   * @param contextInfo - Additional context for error messaging (e.g., question index)
+   */
+  private validateIncludeIfSelectedOptions(
+    includeIfSelected: string[],
+    availableOptions: string[],
+    contextInfo: string,
+  ): void {
+    if (!includeIfSelected?.length) return;
+
+    const invalidOptions = includeIfSelected.filter(
+      (selectedOption) => !availableOptions.includes(selectedOption),
+    );
+
+    if (invalidOptions.length > 0) {
+      throw new BadRequestException(
+        `${contextInfo}: includeIfSelectedOptions contains invalid options: ${invalidOptions.join(
```

nit: if i got `includeIfSelectedOptions` in my error message i'd probably be a little confused what it means - remember these exception messages are what's being sent to the frontend (and displayed directly to the user if you're not careful!)
if you're not planning on ignoring this error message on the frontend, i think we should change it to something more user-friendly - something like "selections to determine sentence inclusion are not valid question options"

---

i admit this is a very hard error to explain concisely, and i don't think my proposal is the best either